### ISSUE TITLE
Impossible to log in to Twitch

### ISSUE THREAD

**Issue by ruihildt** [mullvad/mullvad-browser#96]
When using the Mullvad browser, I am unable to log in as it states it is
unsupported.

On their website they say to contact the browsers support team for issues
related to non-supported browsers and as I could not find this issue being
discussed elsewhere on the web I figured I'd email you to make you aware.

> Your browser is not currently supported. Please use a recommended browser or learn more here.

**Comment by ruihildt** [comment]
So theoretically, if I set `privacy.resistFingerprinting` to `false`and restart, you should be able to login to Twitch to Mullvad Browser?

Note for lurkers: doing this completely defeats the privacy model of Mullvad Browser, don't do this! This question is just for debugging purposes.

**Comment by Thorin-Oakenpants** [comment]
That's what others have done at arkenfox (note we sanitize on close but AF users can retain site data via site exceptions since we're not in PB Mode - thus no need to login in each session) - i.e they flip RFP, reload and login, and then flip RFP back on

RFP is a global switch and you get can/will get weird behavior/results unless you reload sites - it's not meant to be a toggle, that's what the targets are for. And flipping it will affect every other open window/tab
- so yeah ... **DON'T DO THIS**

if you really need twitch, until and if we can ever resolve this, I suggest you just watch it in a secondary browser, you know, like Firefox - which has partitioning etc, and you're already logging in so it's not a huge deal

**Comment by ruihildt** [comment]
Do you think it's because UTC is used? Would using a timezone that actually exist (for example one of the country that uses GMT all year long) would solve the bug and be an acceptable choice to make?

**Comment by Thorin-Oakenpants** [comment]
I'm still a little wary of that result, but only because I didn't do the testing :) There's elements such as clearing all site data, or perhaps a flagged IP for extra scrutiny, that could effect the result.

I'd like to see someone set their system TZ to UTC0 and try and login on Firefox with a new profile (no extensions or pref flips)

---

> Would using a timezone that actually exist

Well, UTC exists :) It's right there in the standards. If this is the actual cause then it smacks of overzealous anti-bot protection and discrimination against TB to be honest

UTC is unique (well, along with Factory), so there is no direct substitute
- https://arkenfox.github.io/TZP/tests/timezones.html
- ![UTC](https://github.com/mullvad/mullvad-browser/assets/16656956/24e4a7c0-0753-40bd-9168-0af87c40b9b6)

It's the only one that is consistent all year round. Note this is using gecko's supported timezones (`supportedValuesOf`).

If I uncheck that option then we get a larger set that match
- ![expanded](https://github.com/mullvad/mullvad-browser/assets/16656956/6642432e-23a9-40cf-ba63-a6dd463145e9)

But as I said, these aren't "supported" (outside of Factory) in gecko (I assume FF will just report what the OS says), and none are "supported" in chrome for example. So again, they will just stick out, IIUIC, and get flagged again when the bot catches up.

You could always test if GMT works (I don't have a twitch account), but I think it's a losing cause

@tomrittervg 

---

sidenote: if we restrict dates to say 2000+ (so we allow for some backwards compat for calendars) then the diffs fall away and we have these matching in supportedValuesOf (most differences are due to old-timey changes)
```
Africa/Abidjan
Africa/Accra
Africa/Bamako
Africa/Banjul
Africa/Bissau
Africa/Conakry
Africa/Dakar
Africa/Freetown
Africa/Lome
Africa/Monrovia
Africa/Nouakchott
Africa/Ouagadougou
Africa/Timbuktu
America/Danmarkshavn
Atlantic/Reykjavik
Atlantic/St_Helena
Factory
UTC
```

So a case could be made to be `Atlantic/Reykjavik` for all users, for example - it would really affect 99.9% of use cases different to UTC

**Comment by ruihildt** [comment]
I meant exist, as being a recognizable named timezone. :) 

Which arguably would be impossible to use a blocking parameter, since a whole "legitimate" named timezone would be blocked. But if they purposefully block Tor Browser, then it's no use indeed. It would be worth it if it was to defeat some bots which uses UTC by default.

**Comment by Thorin-Oakenpants** [comment]
The reason for UTC is it's stable all year round, so users can always work out their real time from +/- from UTC. But I doubt any OS uses UTC or GMT for that matter as their TZ, so there is no solution short of using some thing like `Atlantic/Reykjavik` or `- `Africa/Dakar` ... which _are_ used, so yes, I get your point

I will see at what point in time any of those differed:

OK, these
 - Africa/Abidjan, Africa/Accra, Africa/Bamako, Africa/Banjul, Africa/Conakry, Africa/Dakar, Africa/Freetown, Africa/Lome, Africa/Nouakchott, Africa/Ouagadougou, Africa/Timbuktu, Atlantic/Reykjavik, Atlantic/St_Helena
 - ^ are identical to UTC except before 1911, where they differ by 16.133333 minutes

**Comment by Thorin-Oakenpants** [comment]
so I would argue that yes we could use one of these instead and it wouldn't make any difference to today's usage of UTC, but it may be confusing to users - depends how a website shows dates/times

**Comment by ruihildt** [comment]
So I tested if the timezone was the only discriminating factor, by testing those two scenarios:
- Test 1: creating a new firefox profile, connection without VPN, passing the UTC timezone to the Firefox process (See: https://stackoverflow.com/a/26100074)
- Test 2: creating a new profile, connection without VPN, using my default system timezone

And I was blocked in Test 1 but not in Test 2.^^"

**Comment by Thorin-Oakenpants** [comment]
> passing the UTC timezone to the Firefox process at launch time (See: https://stackoverflow.com/a/26100074)

In windows, I just change the TZ via system settings and restart FF :)

**Comment by Thorin-Oakenpants** [comment]
https://gitlab.torproject.org/tpo/applications/mullvad-browser/-/issues/189

**Comment by Thorin-Oakenpants** [comment]
thanks, that pretty much confirms it's at least `UTC` - I wonder if it does the same with Factory or GMT etc, not that it matters

**Comment by ruihildt** [comment]
It's probably UTC+Linux.

If GMT or another named timezone works, do you see a downside to try it for a specific release?

**Comment by Thorin-Oakenpants** [comment]
GMT is not a `supportedValueOf` in gecko so I'm not sure if that will affect anything, but otherwise it is an exact match for UTC. The problem is that it isn't used in real-world (IIUIC) so we will only end up in an arms race with the anti-bot script

Someone with a twitch account should check windows - I don't think it's linux (or mac) specific. Lots of ways to test that, but it doesn't matter because we should use the same TZ for all platforms for simplicity

**Comment by Thorin-Oakenpants** [comment]
both TZ  and mismatched userAgent-vs-header are resolved in 13.5 - both of which "helped" cause twitch to throw hissy fits
